### PR TITLE
App Log part 3 Recycler

### PR DESCRIPTION
**Pull request by ricdezen**
### Irrelevant info
Hi. If this is badly formatted I apologize, I'm at a friends house after seeing the Sonic movie. Went just for Jim Carrey, worth it.

## Premise

This pull request follows part 2, so check that out before this otherwise you'll just see all the changes together since this is a direct continuation of the previous pull.
This is the third part of the log features. The recycler adapter and viewholder.

## Changes

- Added `LogItem` as a very simple pojo for the data that should be displayed in the log, they hold strings without any particular checks, and hold a boolean field to track their "state", this state is the "expansion". Basically it means the item should be displayed as expanded or not (and such state changes on touch).
- Added `LogRecyclerAdapter` and `LogViewHolder`, they don't really do much, they just load the data from a LogItem list and display it.
- Added `LogItemFormatter` as a class to build the LogItems from a `SMSLogEvent`. Name and Address are left as is, the time is converted from long to String date (I have to change this into formatting as hh:mm when the time refers to the current day, it's fairly easy)

## Side Changes

- Ok my bad I was wrong the sms contact generator was in part 4.
- Removed the empty example tests to remove codacy warnings.

### Footer

Everybody knows Fennecs, but here are their cousins the [Bengala foxes](https://images.app.goo.gl/ZopAqj54Fat9Qk6q8)

### CODE REVIEW

**Comment by RicDeZen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/test/java/ingsw/group1/findmyphone/event/SMSLogDatabaseTest.java  4
- app/src/main/java/ingsw/group1/findmyphone/fragment/LogFragment.java  1
- app/src/main/java/ingsw/group1/findmyphone/log/LogItemFormatter.java  4
- app/src/test/java/ingsw/group1/findmyphone/log/LogItemTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/log/LogItem.java  1
- app/src/androidTest/java/ingsw/group1/findmyphone/LogFragmentTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/event/SMSLogDatabase.java  4
- app/src/main/java/ingsw/group1/findmyphone/log/LogViewHolder.java  4
- app/src/test/java/ingsw/group1/findmyphone/RandomSMSContactGenerator.java  2
- app/src/main/java/ingsw/group1/findmyphone/log/LogRecyclerAdapter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5009715&bid=16524546)

File: app/src/main/java/ingsw/group1/findmyphone/log/LogItem.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+/**
+ * Class representing the data for an item view in the log list.
+ * Compared to {@link SMSLogEvent} the data should already be formatted for display here.
+ * This is a simple pojo, no logic is contained here.
+ *
+ * @author Riccardo De Zen.
+ * @see LogItemFormatter for details on item formatting.
+ */
+public class LogItem {
```
Comment (turcato-niccolo) [inline app/src/main/java/ingsw/group1/findmyphone/log/LogItem.java:17]: So these objects should contain the drawables to display?

File: app/src/main/java/ingsw/group1/findmyphone/log/LogItemFormatter.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+     * @param eventToFormat The event to format. Unknown type events can not be parsed through
+     *                      this method.
+     * @return The formatted {@link LogItem}. Returns {@code null} if the event is of an unknown
+     * type or if the drawable icon could not be loaded.
+     */
+    @Nullable
+    public LogItem formatItem(SMSLogEvent eventToFormat) {
+        if (eventToFormat.getType() == LogEventType.UNKNOWN)
+            return null;
+        Drawable appropriateDrawable = cachedDrawables.get(eventToFormat.getType());
```
Comment (turcato-niccolo) [inline app/src/main/java/ingsw/group1/findmyphone/log/LogItemFormatter.java:54]: Shouldn't you create a new drawable for each object? I've not used these yet, so I'm asking if using the same references for more objects will create problems

File: app/src/main/java/ingsw/group1/findmyphone/log/LogItem.java (hunk 1 above)
Comment (ricdezen) [reply app/src/main/java/ingsw/group1/findmyphone/log/LogItem.java:17]: Yes they hold directly the reference to the appropriate drawable, having matching data and icon is responsibility of the formatter

File: app/src/main/java/ingsw/group1/findmyphone/log/LogItemFormatter.java (hunk 2 above)
Comment (ricdezen) [reply app/src/main/java/ingsw/group1/findmyphone/log/LogItemFormatter.java:54]: Worked as far as I could tell, Drawables are kinda like an in memory representation of an image file, so they can be shared

Comment (turcato-niccolo) [reply app/src/main/java/ingsw/group1/findmyphone/log/LogItemFormatter.java:54]: Nice

**Review by turcato-niccolo — DISMISSED** [review]
Looking good

**Comment by ricdezen** [comment]
Wait wait gotta pull from development first, still got time to screw up.

**Comment by ricdezen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/log/LogItemFormatter.java  4
- app/src/main/java/ingsw/group1/findmyphone/fragment/LogFragment.java  1
- app/src/test/java/ingsw/group1/findmyphone/log/LogItemTest.java  1
- app/src/androidTest/java/ingsw/group1/findmyphone/LogFragmentTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/log/LogRecyclerAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/log/LogViewHolder.java  4
- app/src/main/java/ingsw/group1/findmyphone/log/LogItem.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5009715&bid=16524546)

**Comment by ricdezen** [comment]
Pulled from development, fixed renaming of EventType. Anyone willing to check out the result can run the `LogFragmentTest` inside the AndroidTest folder. Just runs the log fragment for a bunch of seconds allowing to interact with it. Provides an example of every item type, plus a failed operation to show it's properly marked in red and won't expand.

P.S. To answer Niccolò's question from earlier, two items easily share the same image in the test.

**Comment by turcato-niccolo** [comment]
> Pulled from development, fixed renaming of EventType. Anyone willing to check out the result can run the `LogFragmentTest` inside the AndroidTest folder. Just runs the log fragment for a bunch of seconds allowing to interact with it. Provides an example of every item type, plus a failed operation to show it's properly marked in red and won't expand.

Tests successful on my machine

> P.S. To answer Niccolò's question from earlier, two items easily share the same image in the test.

Does it work as you wanted?

**Comment by ricdezen** [comment]
> Does it work as you wanted?

Meh I dunno it's alright

**Review by turcato-niccolo — APPROVED** [review]

**Pull request merged**